### PR TITLE
Enhancements for RANGE tests

### PR DESCRIPTION
**Pull request by evgeniiz321**

### CODE REVIEW

**Review by roman-khimov — COMMENTED** [review]
Range overflow test?

File: pytest_tests/tests/object/test_object_api.py [hunk 1]
```diff
@@ -592,6 +592,52 @@ def test_object_get_range(self, request: FixtureRequest, storage_objects: list[S
...
+
+        file_ranges_to_test = []
+
+        parts = get_object_chunks(default_wallet.path, container, oid, self.shell, self.neofs_env)
+
+        # range is inside one child
+        file_ranges_to_test.append((0, parts[0][1] - 1))
+        # range matches child
+        file_ranges_to_test.append((parts[0][1], parts[1][1]))
+        # range requires more than one child (it includes OR does not include the first and the last children)
```
Comment (carpawell) [inline pytest_tests/tests/object/test_object_api.py:618]: in the issue, i meant both cases (a one when range includes end pieces and a one when does not). currently, this comment sounds like it does not know what is happening

File: pytest_tests/tests/object/test_object_api.py [hunk 2]
```diff
@@ -592,6 +592,52 @@ def test_object_get_range(self, request: FixtureRequest, storage_objects: list[S
...
+        file_ranges_to_test.append((0, parts[0][1] + parts[1][1] - 1))
+        # range requires more than two children (it includes OR does not include the first and the last children)
+        file_ranges_to_test.append((0, complex_object_size - 1))
+        file_ranges_to_test.append((parts[0][1] + 1, complex_object_size - parts[0][1] - parts[-1][1] - 1))
```
Comment (carpawell) [inline pytest_tests/tests/object/test_object_api.py:622]: looks working correctly but really relies on the fact that there will always be more than two parts, not sure if it is error-proof

**Comment by carpawell** [comment]
> Range overflow test?

As I remember there are some tests about it already, but @evgeniiz321, can prove it.

File: pytest_tests/tests/object/test_object_api.py (hunk 2 above)
Comment (evgeniiz321) [reply pytest_tests/tests/object/test_object_api.py:622]: With the current settings there always will be more than two parts. If in the future we will update something, we will update this accordingly. Making an error-proof test usually is a waste of time.

**Comment by evgeniiz321** [comment]
> Range overflow test?

Yes, we already have it in negative scenarios.

File: pytest_tests/tests/object/test_object_api.py (hunk 1 above)
Comment (evgeniiz321) [reply pytest_tests/tests/object/test_object_api.py]: I updated comments and cases a little bit

File: pytest_tests/tests/object/test_object_api.py [hunk 3]
```diff
@@ -99,7 +99,7 @@ def generate_ranges(
 
     for offset, length in file_ranges:
         range_length = random.randint(RANGE_MIN_LEN, RANGE_MAX_LEN)
-        range_start = random.randint(offset, offset + length)
+        range_start = random.randint(offset, offset + length - 1)
```
[the remark this answers is not in the document]
Comment (evgeniiz321) [reply pytest_tests/tests/object/test_object_api.py:102]: Please, open a separate issue to enhance the readability and to refactor all the functions/methods/classes that seem not good for you. I know there are some very strange things going in the tests :)

[the remark this answers is not in the document]
Comment (carpawell) [reply pytest_tests/tests/object/test_object_api.py:102]: #807

**Review by carpawell — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**